### ISSUE TITLE
Cross namespaces VirtualServices have a different behavior since Istio 1.12

### ISSUE THREAD

**Issue by djannot** [istio/istio#37691]
### Bug Description

I'm deploying the bookinfo application in 2 different namespaces.
The `productpage` service in the `bookinfo-frontends` namespace.
The other services in the `bookinfo-backends` namespace.

Then, I create the following VirtualService in the `bookinfo-frontends` namespace:

```
apiVersion: networking.istio.io/v1beta1
kind: VirtualService
metadata:
  name: reviews
  namespace: bookinfo-frontends
spec:
  hosts:
    - reviews.bookinfo-backends.svc.cluster.local
  http:
  - route:
    - destination:
        host: reviews.bookinfo-backends.svc.cluster.local
    timeout: 0.5s
```

But it's not applied.

If I run the `istioctl pc route deploy/productpage-v1.bookinfo-frontends --name 9080` command, it confirms that the VirtualService isn't applied:

```
NAME     DOMAINS                                                     MATCH     VIRTUAL SERVICE
9080     details.bookinfo-backends, 10.2.100.220                     /*        
9080     productpage, productpage.bookinfo-frontends + 1 more...     /*        
9080     ratings.bookinfo-backends, 10.2.255.201                     /* 
9080     reviews.bookinfo-backends, 10.2.53.134                      /*
```

Now, if I create the same VirtualService in the `bookinfo-backends` namespace, it's correctly applied.

Here is the output of the same command:

```
NAME     DOMAINS                                                     MATCH     VIRTUAL SERVICE
9080     details.bookinfo-backends, 10.2.100.220                     /*        
9080     productpage, productpage.bookinfo-frontends + 1 more...     /*        
9080     ratings.bookinfo-backends, 10.2.255.201                     /*
9080     reviews.bookinfo-backends, 10.2.53.134                      /*        reviews.bookinfo-backends
```

In previous versions (tested in 1.11.7), it works with the VirtualService in `bookinfo-frontends`.
But in newer versions (tested in 1.12.4 and 1.13.1), it doesn't.

### Version

```prose
$ istioctl version
client version: 1.12.4
control plane version: 1.12.4
data plane version: 1.12.4 (11 proxies)

$ kubectl version --short
Client Version: v1.23.1
Server Version: v1.21.1
```

**Comment by nrjpoddar** [comment]
This feels like a regression that we should have caught in our integration tests.

@howardjohn @stevenctl can you shed more light here?

**Comment by howardjohn** [comment]
Do we know root cause? This seems like an absolutely trivial VS

**Comment by howardjohn** [comment]
I think its broken by https://github.com/istio/istio/commit/dea2be520667359e0f44e7d1db24b01f26b16741

**Comment by howardjohn** [comment]
@hzxuzhonghu

**Comment by howardjohn** [comment]
cc @ramaraochavali as well

**Comment by hzxuzhonghu** [comment]
I think we have integration test covering vs in same.different namespaces as backend service

**Comment by hzxuzhonghu** [comment]
I see.  It looks not right

```
		// Check if there is an explicit import of form ns/* or ns/host
		if importedHosts, nsFound := hosts[configNamespace]; nsFound {
			addVirtualService(c, importedHosts)
		}
```

**Comment by ldemailly** [comment]
So I see this is fixed in 1.12.6, thanks! which 1.13 has the fix?
